src/annotator/guest.js

```diff
@@ -318,10 +323,8 @@ export default class Guest {
           return;
         }
 
-        this._onClearSelection({ informHostFrame: false });
-        // TODO: clear the selection calling `document.getSelection()?.removeAllRanges();`
-        // without triggering the selection observer (that in turns calls
-        // `_onClearSelection({informHostFrame: true})` and changes `Sidebar#._guestWithSelection`).
+        removeTextSelection();
+        this._informHostOnNextSelectionClear = false;
```

Comment:
This ordering works because `selectionchanged` is dispatched asynchronously. Logically I think it would be nicer to set the flag before removing the selection.

Reply:
👍🏼